templates/todo/api/java/src/main/java/com/microsoft/azure/simpletodo/configuration/AzureDeveloperCliCredential.java

```diff
@@ -0,0 +1,220 @@
+     */
+    public AzureDeveloperCliCredential() {
+        identityClientOptions = new IdentityClientOptions();
+    }
+
+    @Override
+    public Mono<AccessToken> getToken(TokenRequestContext request) {
+        return authenticateWithAzureCli(request)
+            .doOnNext(token -> LoggingUtil.logTokenSuccess(LOGGER, request))
+            .doOnError(error -> LoggingUtil.logTokenError(LOGGER, identityClientOptions, request, error));
+    }
+
+    private Mono<AccessToken> authenticateWithAzureCli(TokenRequestContext request) {
+        StringBuilder azCommand = new StringBuilder("azd auth token --output json --scope ");
+
+        var scopes = String.join(" --scope ", request.getScopes());
+
+        try {
+            ScopeUtil.validateScope(scopes);
+        } catch (IllegalArgumentException ex) {
+            return Mono.error(LOGGER.logExceptionAsError(ex));
+        }
+
+        azCommand.append(scopes);
+
+        AccessToken token;
+        try {
+            String starter;
+            String switcher;
+            if (isWindowsPlatform()) {
+                starter = WINDOWS_STARTER;
+                switcher = WINDOWS_SWITCHER;
+            } else {
+                starter = LINUX_MAC_STARTER;
+                switcher = LINUX_MAC_SWITCHER;
+            }
+
+            ProcessBuilder builder = new ProcessBuilder(starter, switcher, azCommand.toString());
+
+            String workingDirectory = getSafeWorkingDirectory();
+            if (workingDirectory != null) {
+                builder.directory(new File(workingDirectory));
+            } else {
+                throw LOGGER.logExceptionAsError(
+                    new IllegalStateException(
+                        "A Safe Working directory could not be" + " found to execute Azure Developer CLI command from."
+                    )
+                );
+            }
+            builder.redirectErrorStream(true);
+            Process process = builder.start();
+
+            StringBuilder output = new StringBuilder();
+            try (
+                BufferedReader reader = new BufferedReader(
+                    new InputStreamReader(process.getInputStream(), StandardCharsets.UTF_8.name())
+                )
+            ) {
+                String line;
+                while (true) {
+                    line = reader.readLine();
+                    if (line == null) {
+                        break;
+                    }
+
+                    if (
+                        line.startsWith(WINDOWS_PROCESS_ERROR_MESSAGE) ||
+                        LINUX_MAC_PROCESS_ERROR_MESSAGE.matcher(line).matches()
+                    ) {
+                        throw LoggingUtil.logCredentialUnavailableException(
+                            LOGGER,
+                            identityClientOptions,
+                            new CredentialUnavailableException(
+                                "AzureCliCredential authentication unavailable. Azure CLI not installed." +
```

Comment:
This needs replacement in a few places... AzureCliCredential -> AzureDeveloperCliCredential